### PR TITLE
10 post add new comment

### PR DESCRIPTION
**Pull request by baldock92**
- Create function to POST new comment by review_id
- Tests written
- Error handling for invalid username, review_id and missing mandatory keys.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -285,3 +285,69 @@ describe("GET - /api/reviews/:review_id/comments", () => {
...
+
+    return request(app)
+      .post("/api/reviews/2/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("User not found");
+      });
+  });
+});
```
Comment (DCrawley94) [inline __tests__/app.test.js:367]: Nicely done, have you thought about what should happen if we make a post request to an invalid id (for example an ID of 'not-an-id'

File: controllers/comments.controller.js [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+  if (
+    req.body.hasOwnProperty("username") === false ||
+    req.body.hasOwnProperty("body") === false
+  ) {
+    res.status(400).send({ msg: "Mandatory field missing" });
+  }
+
+  const promises = [
+    getAllReviewsHere(review_id),
+    checkUserExists(username),
```
Comment (DCrawley94) [inline controllers/comments.controller.js]: So this is a good idea. However we can avoid checking the username manually by just letting Postgres deal with it. What would happen if you removed this?

File: controllers/comments.controller.js [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+    addCommentByReviewId(review_id, username, body),
+  ];
+
+  return Promise.all(promises)
+    .then(([irrelevant1, irrelevant2, comment]) => {
```
Comment (DCrawley94) [inline controllers/comments.controller.js]: Minor comment about the variable names here. I'd recommend using '_' as a variable name as it's commonly used to represent a parameter that isn't going to be used :)

**Review by DCrawley94 — CHANGES_REQUESTED** [review]
Really nice work George! A few minor changes but overall looking really good :)

**Comment by baldock92** [comment]
- Additional error handling test written to check if an invalid review_id is passed.
- Refactor error handling for an invalid username, using psql error codes.
edit:
-Also rename un-needed variable in Promise.all return to "_"

**Review by DCrawley94 — APPROVED** [review]
Looks great George! Feel free to merge :)

**Pull request merged**